File: Python/1.py
# Algorithm
    # Create hash storing index of each number
    # Iterate through nums and check whether target-num in hash
    # Observe that we don't have to store multiple indices because the first time
        # we encounter a pair we'll return the values
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index = {}
        for i, n in enumerate(nums):
            if target - n in index.keys():
                return [index[target - n], i]
            else:
                index[n] = i

File: C++/1.cpp
/*
- Note: better to iterate through in one pass than create map first because
- map.end() can't be compared to an int, so can't check for unique indexes
*/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> mp;

        for(int i = 0; i < n; i++){
            int c = target - nums[i];
            if(mp.find(c) != mp.end()){
                return {mp[c], i};
            }
            mp.insert({nums[i], i});
        }
        return {};
    }
};